server/controllers/product.js

```diff
@@ -1,15 +1,34 @@
-import ProductModel from '../models/Product';
+import { query } from "express";
+import { reset } from "nodemon";
+import ProductModel from "../models/Product";
 
 const list = (req, res) => {
-  //logic for get a list of all products from mongodb here
-  const products = [
-    { id: 1, name: 'computadora noganet' },
-    { id: 2, name: 'mouse steelseries' },
-    { id: 3, name: 'monitor benq' },
-  ];
-  res.send(products);
+  const name = req.query.name;
+  const category = req.query.category;
```

Comment:
esto lo podes reescribir mas chetamente de esta forma


```suggestion
  const { name, category } = req.query;
```